### PR TITLE
Xdog expr and cyto

### PR DESCRIPTION
**Pull request by Afollet**
Add cytogenetic and expression data types to the xdog. Updated how xdog extracts to work with the updated data model. Added Entrez2Hugo gene converter. Adds NCBI gene Id to markers for extraction. Modified test in exportDataService and cbioportal transformer

### CODE REVIEW

File: indexer/src/main/java/org/pdxfinder/utils/CbpTransformer.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    public enum cbioType {
+        MUT,
+        GISTIC
+    }
+
+    public void exportCBP(File exportDir,File templateDir, File pathToJson, cbioType dataType) throws IOException {
+
+        if (doesFileNotExist(exportDir) || doesFileNotExist(templateDir) || doesFileNotExist(pathToJson)) {
+            throw new IOException("A string argument passed to the exportCBP does not point to an existing file.");
```
Comment (rmwthorne) [inline indexer/src/main/java/org/pdxfinder/utils/CbpTransformer.java:50]:
```suggestion
            throw new IOException("A string argument passed to the exportCBP does not point to an existing file:" +
            "{}%n{}%n{}%n", exportDir, templateDir, pathToJson);
```

File: indexer/src/main/java/org/pdxfinder/utils/CbpTransformer.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            cbpMapsToSheetsByDataType(listMapTable, dataType);
+
+            universalDataExporter.setDs(jsonGroup);
+            universalDataExporter.setTemplateDir(templateDir.getAbsolutePath());
+            universalDataExporter.export(exportDir.getAbsolutePath());
+    }
+
+    public void convertListOfEntrez(List<String> entrezIds){
+        entrezIds.forEach(e ->
+            log.info(omicTransformationService.ncbiGeneIdtoHgncSymbol(e))
```
Comment (rmwthorne) [inline indexer/src/main/java/org/pdxfinder/utils/CbpTransformer.java:64]: This method does not appear to do anything other than log - you want to clarify the intention in the method name or perform the functionality outside of the `log.info()` call

File: indexer/src/test/java/org/pdxfinder/commandline/FinderCommandLineTest.java [hunk 3]
```diff
@@ -60,22 +63,44 @@ public void setUp() {
     }
 
     @Test public void load_givenTransform_When_exportAllisCalled_Then_callsTransformer() throws IOException {
-        String[] args = {"--data-dir=path/", "--all"};
+        String[] args = {"--data-dir=path/", "-c=mut"};
+        int exitCode = new CommandLine(transform).execute(args);
+        assertEquals(0, exitCode);
+        verify(this.finderTransformer).run(
+                any(File.class),
+                any(null),
```
Comment (rmwthorne) [inline indexer/src/test/java/org/pdxfinder/commandline/FinderCommandLineTest.java:71]: Replace all instances of `any(null)` with  `any()` - you shouldn't pass `null` as a parameter to methods

File: indexer/src/test/java/org/pdxfinder/commandline/FinderCommandLineTest.java [hunk 4]
```diff
@@ -60,22 +63,44 @@ public void setUp() {
     }
 
     @Test public void load_givenTransform_When_exportAllisCalled_Then_callsTransformer() throws IOException {
```
Comment (rmwthorne) [inline indexer/src/test/java/org/pdxfinder/commandline/FinderCommandLineTest.java:65]: Remove `load` from method name. This comes from other tests that use the naming convention `methodNameUnderTest_givenCondition_thenBehavesCertainWay`. It's a good time to review the names for the other tests in this file too.

File: indexer/src/test/java/org/pdxfinder/commandline/FinderExporterTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    @SpyBean
+    FinderExporter finderExporter;
+
+    @Before public void init() throws IOException {
+        tempFile = folder.newFile();
+    }
+
+    @Test
+    public void Given_loadAll_CallExportAll() throws IOException {
```
Comment (rmwthorne) [inline indexer/src/test/java/org/pdxfinder/commandline/FinderExporterTest.java:39]: Method names should follow the existing convention formatting

File: data-services/src/main/java/org/pdxfinder/services/loader/envload/LoadMarkers.java [hunk 6]
```diff
@@ -91,34 +99,31 @@ public void loadGenes(String dataURL) {
...
-                        Marker m = new Marker();
-                        m.setHgncSymbol(symbol);
-
-                        if (!ensemblId.isEmpty()) {
-                            m.setEnsemblGeneId(ensemblId);
-                        }
-                        if (!hgncId.isEmpty()) {
-                            m.setHgncId(hgncId);
-                        }
+                    Marker m = new Marker();
```
[the remark this answers is not in the document]
Comment (Afollet) [reply data-services/src/main/java/org/pdxfinder/services/loader/envload/LoadMarkers.java:102]: Should I pass the variables are arguments in the new method?

**Pull request merged**